button/internal/button_test.ts

```diff
@@ -0,0 +1,215 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import '../text-button.js';
+
+import {html, render} from 'lit';
+
+describe('Button', () => {
+  let root: HTMLDivElement;
+
+  beforeEach(() => {
+    root = document.createElement('div');
+    document.body.appendChild(root);
+  });
+
+  afterEach(() => {
+    root?.remove();
+  });
+
+  it('renders inner button', async () => {

```

Comment:
These tests seem really granular and test the implementation rather than the functionality.

What sort of value are we hoping to get out of these tests?

Reply:
we don't seem to have any button tests yet.

i agree these are fairly granular as they are basically asserting we rendered what we expected to render.

you could try test from the outside instead, e.g. test that a button with a `href` is clickable as a link. though that seems we will just end up back here with less assertions (but the same remaining assertions).

one way or another, in that example, you have to test that the `href` actually resulted in something. so you have to assert we opened the link after clicking the button, or that the link existed in DOM.